ocsp/responder/db_source_test.go

```diff
@@ -0,0 +1,176 @@
+package responder
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/go-gorp/gorp/v3"
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/db"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+	"github.com/letsencrypt/boulder/test"
+	"golang.org/x/crypto/ocsp"
+)
+
+// To mock out WithContext, we need to be able to return objects that satisfy
+// gorp.SqlExecutor. That's a pretty big interface, so we specify one no-op mock
+// that we can embed everywhere we need to satisfy it.
+// Note: mockSqlExecutor does *not* implement WithContext. The expectation is
+// that structs that embed mockSqlExecutor will define their own WithContext
+// that returns a reference to themselves. That makes it easy for those structs
+// to override the specific methods they need to implement (e.g. SelectOne).
+type mockSqlExecutor struct{}
+
+func (mse mockSqlExecutor) Get(i interface{}, keys ...interface{}) (interface{}, error) {
+	return nil, fmt.Errorf("unimplemented")
```

Comment:
Nit: I know this line (and those below this line) actually came from original source but we might also refactor these `fmt.Errorf` calls to `errors.New`